bin/pt-replica-find

```diff
@@ -0,0 +1,4819 @@
+   # If --pid, check it first since we'll die if it already exits.
+   # ########################################################################
+   my $daemon;
+   if ( $o->get('pid') ) {
+      # We're not daemoninzing, it just handles PID stuff.  Keep $daemon
+      # in the the scope of main() because when it's destroyed it automatically
+      # removes the PID file.
+      $daemon = new Daemon(
+         daemonize => 0, # not daemoninzing, just PID file
+         pid_file => $o->get('pid'),
+      );
+      $daemon->run();
+   }
+
+   # ########################################################################
+   # Connect and go to work.
+   # ########################################################################
+   my $dbh = $dp->get_dbh($dp->get_cxn_params($master_dsn), {AutoCommit => 1});
+
+   my $exit_status = 0;
+   my %slaves_found;
+   my $root;
+
+   # Despite the name, recursing to slaves actually begins at the specified
+   # server, so the named server may also be included.
+   my $ms = new MasterSlave(
+      OptionParser => $o,
+      DSNParser    => $dp,
+      Quoter       => "Quoter",
+   );
+   $ms->recurse_to_replicas(
+      {  dbh      => $dbh,
+         dsn      => $master_dsn,
+         replica_user     => ( $o->got('replica-user') or $o->got('slave-user') ) ? $o->get('replica-user') : '',
+         replica_password => ( $o->got('replica-password') or $o->got('slave-password') ) ? $o->get('replica-password') : '',
+         callback => sub {
+            my ( $dsn, $dbh, $level, $parent ) = @_;
+            if ( !$parent ) {
+               $root = $dsn;
+            }
+            else {
+               my $p = $slaves_found{$dp->as_string($parent)};
+               push @{$p->{children}}, $dsn;
+            }
+            my $slave = $dp->as_string($dsn);
+            $slaves_found{$slave}        = $dsn;
+            $slaves_found{$slave}->{dbh} = $dbh;
+         },
+      }
+   );
+
+   # If the user relied on values from the defaults file, the $root hostname
+   # might not be defined.  Try to fill it in.
+   $root->{h} ||= $dp->get_hostname($dbh);
+
+   # Now that we've found them all, recurse through them and do stuff!
+   my %print_node_sub_for = (
+      hostname => \&print_node_hostname,
+      summary  => \&print_node_summary,
+   );
+   my $print_node = $print_node_sub_for{lc($o->get('report-format'))};
+   print_slaves(
+      node          => $root,
+      print_node    => $print_node,
+      MasterSlave   => $ms,
+      resolve_address => $o->get('resolve-address'),
+   );
+
+   return 0;
+}
+
+# ############################################################################
+# Subroutines
+# ############################################################################
+
+# Print master and slaves in a tree like:
+#    master:port
+#    +- slave1:port
+#       +- slave1-1:port
+#    +- slave2:port
+# The levels are thus:
+#    0
+#    +- 1
+#       +- 2
+#    +- 1
+sub print_slaves {
```

Comment:
Since many internal variables were renamed in lib, I'm commenting to be thorough
This tool package did not receive the s/slave/replica
However I don't see any misusage

Unless this was a goal to remove all of them in this PR specifically, I'm ok to keep it and maybe flag it for later